src/Microsoft.ML.Dnn/DnnCatalog.cs

```diff
@@ -80,59 +81,53 @@ public static class DnnCatalog
-            {
-                FeaturesColumnName = featuresColumnName,
-                LabelColumnName = labelColumnName,
-                ScoreColumnName = scoreColumnName,
-                PredictedLabelColumnName = predictedLabelColumnName,
-                ValidationSet = validationSet
-            };
+            options.EarlyStoppingCriteria = options.DisableEarlyStopping ? null : options.EarlyStoppingCriteria ?? new ImageClassificationTrainer.EarlyStopping();
 
-            return ImageClassification(catalog, options);
+            var env = CatalogUtils.GetEnvironment(catalog);
+            return new ImageClassificationTrainer(env, options);
         }
 
         /// <summary>
         /// Performs image classification using transfer learning.
-        /// usage of this API requires additional NuGet dependencies on TensorFlow redist, see linked document for more information.
+        /// Usage of this API requires additional NuGet dependencies on TensorFlow redist, see linked document for more information.
         /// <format type="text/markdown">
         /// <![CDATA[
         /// [!include[io](~/../docs/samples/docs/api-reference/tensorflow-usage.md)]
         /// ]]>
         /// </format>
         /// </summary>
-        /// <param name="catalog"></param>
-        /// <param name="options">An <see cref="Options"/> object specifying advanced options for <see cref="ImageClassificationEstimator"/>.</param>
-        public static ImageClassificationEstimator ImageClassification(
-            this ModelOperationsCatalog catalog, Options options)
-        {
-            options.EarlyStoppingCriteria = options.DisableEarlyStopping ? null : options.EarlyStoppingCriteria ?? new EarlyStopping();
+        /// <param name="catalog">Catalog</param>
+        /// <param name="labelColumnName">The name of the labels column.</param>
+        /// <param name="featureColumnName">The name of the input features column.</param>
+        /// <param name="scoreColumnName">The name of the output score column.</param>
+        /// <param name="predictedLabelColumnName">The name of the output predicted label columns.</param>
+        /// <param name="validationSet">The validation set used while training to improve model quality.</param>
 
-            var env = CatalogUtils.GetEnvironment(catalog);
-            return new ImageClassificationEstimator(env, options, DnnUtils.LoadDnnModel(env, options.Arch, true));
+        public static ImageClassificationTrainer ImageClassification(
+            this MulticlassClassificationCatalog.MulticlassClassificationTrainers catalog,
+            string labelColumnName = DefaultColumnNames.Label,
+            string featureColumnName = DefaultColumnNames.Features,
+            string scoreColumnName = DefaultColumnNames.Score,
+            string predictedLabelColumnName = DefaultColumnNames.PredictedLabel,
+            IDataView validationSet = null)
```

Comment:
I thought you were going to move the `validationSet` to be during the `Fit` method?

Reply:
I have and it works well when the trainer is the last item in the pipeline but if it is not then the pipeline becomes nasty and you will need to save the model twice. Most people would want to add KeyToValue Transform after this trainer to convert predictedLabel to string type for example .

---
In reply to: [338737502](https://github.com/dotnet/machinelearning/pull/4372#discussion_r338737502) [](ancestors = 338737502)